scripts/export_environment.py

```diff
@@ -264,19 +264,42 @@ def resolve_env(env: Environment) -> Environment:
     return dict(ResolvedEnvironment(env))
 
 
+azul_env_vars = 'AZUL_ENV_VARS'
+
+
 def export_env(env: Environment, output: Optional[TextIO]) -> None:
     """
-    Print the given environment in a form that can be evaluated by a shell.
+    Print the given environment in a form that can be evaluated by a shell,
+    unsetting variables that are no longer used.
     """
+    try:
+        old_vars = os.environ[azul_env_vars]
+    except KeyError:
+        old_vars = set()
+    else:
+        old_vars = set(old_vars.split(','))
+    assert not any(',' in var for var in env), env
+    env = {
+        azul_env_vars: ','.join(env.keys()),
+        **env
+    }
+    for unset in old_vars - env.keys():
+        print(f"{this_module.name}: {'Would unset' if output is None else 'Unsetting'} "
+              f"{unset}",
+              file=sys.stderr)
+        if output is not None:
+            print(f'unset {unset}', file=output)
```

Comment:
This works? Don't you want `export -n`?

Reply:
This works. `export -n` seems to be bash specific, where `unset` is more portable. For example `export -n` doesn't seem to work in zsh which is the default shell for macOS.

Reply:
Bash is a requirement of Azul. In general, we do not aim to support other shells.

I looked into this and it seems that for bash, `unset FOO` is the inverse of `export FOO=bar`. `export -n` is wrong since it leaves the shell variable set:

```
hannes@hannes-xps:~$ env | grep FOO
hannes@hannes-xps:~$ export FOO=bar
hannes@hannes-xps:~$ env | grep FOO
FOO=bar
hannes@hannes-xps:~$ set | grep FOO
FOO=bar
_=FOO=bar
hannes@hannes-xps:~$ export -n FOO
hannes@hannes-xps:~$ env | grep FOO
hannes@hannes-xps:~$ set | grep FOO
FOO=bar
_=FOO
```

There is one use of `export -n` in Azul. Could you fix that, please?